libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -101,12 +104,12 @@ proc rebalanceMesh(g: GossipSub, topic: string) {.async.} =
     if g.mesh.getOrDefault(topic).len < GossipSubDlo:
       trace "replenishing mesh", topic
       # replenish the mesh if we're below GossipSubDlo
-      while g.mesh.getOrDefault(topic).len < GossipSubD:
+      while g.mesh.getOrDefault(topic).len < GossipSubD and topic in g.topics:
```

Comment:
this was one of the issues.. not checking "in topics" meant the peer will be lost... we would not publish to them

Reply:
A good check, but I doubt that's was causing any issues, the only places where `rebalanceMesh` is called from are:
- https://github.com/status-im/nim-libp2p/blob/ccb323ad0f0da8dfe016e8567efcd5f96f0f3e91/libp2p/protocols/pubsub/gossipsub.nim#L227
- https://github.com/status-im/nim-libp2p/blob/ccb323ad0f0da8dfe016e8567efcd5f96f0f3e91/libp2p/protocols/pubsub/gossipsub.nim#L282
- https://github.com/status-im/nim-libp2p/blob/ccb323ad0f0da8dfe016e8567efcd5f96f0f3e91/libp2p/protocols/pubsub/gossipsub.nim#L449

Which add topics to the `topics` table before calling the `rebalanceMesh` proc.

Reply:
Yeah honestly I'm not sure, but definitely it might have caused some missed publish to fanout (which not sure matters much for NBC), anyway worth testing this!